src/core/core.helpers.js

```diff
@@ -73,21 +73,31 @@ module.exports = function() {
 		var rounded = Math.round(x);
 		return ((rounded - epsilon) <= x) && ((rounded + epsilon) >= x);
 	};
-	helpers.max = function(array) {
-		return array.reduce(function(max, value) {
-			if (!isNaN(value)) {
-				return Math.max(max, value);
-			}
-			return max;
-		}, Number.NEGATIVE_INFINITY);
+	helpers._minAndMax = function(array) {
+		var min = Number.POSITIVE_INFINITY;
+		var max = Number.NEGATIVE_INFINITY;
+		var i, ilen, value;
+
+		for (i = 0, ilen = array.length; i < ilen; i++) {
+			value = array[i];
+			min = Math.min(min, value);
+			max = Math.max(max, value);
+		}
+
+		return {min: min, max: max};
```

Comment:
This would help removing the immediate assignment lines too.
```js
helpers._minAndMax = function(array, target) {
	var min = target ? target.min : Number.POSITIVE_INFINITY;
	var max = target ? target.max : Number.NEGATIVE_INFINITY;
	var i, ilen, value;

	for (i = 0, ilen = array.length; i < ilen; i++) {
		value = array[i];
		min = Math.min(min, value);
		max = Math.max(max, value);
	}

	return helpers.extend(target || {}, {min: min, max: max});
}
`